packages/webapp/src/components/Animals/Inventory/index.tsx

```diff
@@ -83,71 +83,118 @@ const PureAnimalInventory = ({
   view?: View;
 }) => {
   const { t } = useTranslation();
-  const isTaskView = view === View.TASK;
+
+  const viewConfig = () => {
+    switch (view) {
+      case View.TASK:
+        return {
+          tableMaxHeight: !isDesktop || !containerHeight ? undefined : containerHeight - usedHeight,
+          tableSpacerRowHeight: 0,
+          showInventorySelection: isAdmin,
+          showSearchBarAndFilter: true,
+          alternatingRowColor: true,
+          showTableHeader: isDesktop,
+          showActionFloaterButton: false,
+        };
+      case View.TASK_SUMMARY:
+        return {
+          tableMaxHeight: undefined,
+          tableSpacerRowHeight: 0,
+          showInventorySelection: false,
+          showSearchBarAndFilter: false,
+          alternatingRowColor: isDesktop ? false : true,
+          showTableHeader: false,
+          extraRowSpacing: isDesktop,
+          showActionFloaterButton: false,
+        };
+      default:
+        return {
+          tableMaxHeight: !isDesktop || !containerHeight ? undefined : containerHeight - usedHeight,
+          tableSpacerRowHeight: isDesktop ? 96 : 120,
+          showInventorySelection: isAdmin,
+          showSearchBarAndFilter: true,
+          alternatingRowColor: true,
+          showTableHeader: isDesktop,
+          showActionFloaterButton: isAdmin,
+        };
+    }
+  };
```

Comment:
Yeah I hated this part of it the solution. Good call out. I will do that first suggestion.

My instinct was to make more containers and bundle the reused logic into a hook -- but it seemed liek it would take too long. But ideally we would reuse components and not containers. That would make components like KPI and FloatingActionBar not even be possible for task animals Inventory.

Reply:
So I am still not happy with the solution, but I think it is at least better if you could check it out.